### PR TITLE
Split CGALConfig.cmake and CGALExports.cmake

### LINKED ISSUE #471: Debian package libcgal-qt5-dev required when using libcgal-dev with cmake
Original bug report:
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=804539

Background information:
The Debian packages for CGAL enable the additional CGAL componts Qt5, ImageIO, and Core (the latter two are not relevant here). The headers and libraries are put into packages libcgal11v5/libcgal-dev, with the exception of all Qt stuff which goes into its own pair of packages libcgal-qt5-11/libcgal-qt5-dev. This split was done to avoid that all reverse dependencies of CGAL need to depend on the Qt stack, even if they do not need it at all.

Using libcgal11v5/libcgal-dev without having libcgal-qt5-11/libcgal-qt5-dev installed does no longer work in 4.7. It definitely worked in 4.5, and according to the bug submitter also in 4.6.

Note: The examples are a better repro case than the demos since there it is clear that they actually do not need Qt.

Is there a way to support this use case again?

### PR DESCRIPTION
**Pull request by bo0ts**
This PR splits the `CGALConfig.cmake` file into `CGALConfig.cmake` and `CGAL${LIBRARY_NAME}LibConfig.cmake` (N.B. the extra word `Lib` is unfortunately necessary to distinguish the main config file from the config file for the CGAL base library) and generates installations with export files for each library. CMake binary directories still use a single `CGALExport.cmake` file.

I checked the use-case of Debian and performed an installation and removed all libraries besides CGAL and the associated files. The resulting installation works well on it's own. 

Also includes a global removal of `CMAKE_ALLOW_LOOSE_LOOP_CONSTRUCTS`.

Fixes #471

### CODE REVIEW

**Comment by sloriot** [comment]
Merged in CGAL-4.8-Ic-137

**Comment by sloriot** [comment]
At least on windows with cmake  3.3.1 we have the following warning:
`CMake Warning (dev) at D:/CGAL/test/CGAL-4.8-Ic-137/cmake/platforms/x64_Cygwin-Windows8_MSVC2015-Debug-64bits/CGALConfig.cmake:122 (if):
  Policy CMP0054 is not set: Only interpret if() arguments as variables or
  keywords when unquoted.  Run "cmake --help-policy CMP0054" for policy
  details.  Use the cmake_policy command to set the policy and suppress this
  warning.`

**Comment by lrineau** [comment]
I think that you be carefully tested and then pushed into CGAL-4.7.1.

**Comment by bo0ts** [comment]
The new commit should fix the warning.

**Comment by sloriot** [comment]
Merged in CGAL-4.8-Ic-139

**Comment by sloriot** [comment]
@lrineau testsuite shows no issue. Let me know if you want me to rebase on 4.7 and merged it when you're done with you extra tests

**Comment by lrineau** [comment]
I think that should be rebased on CGAL-4.7, if possible. As for the tests, the usual tests are good enough:
- build a demo from the Git layout,
- create a release tarball and try to build a demo from the tarball.
- then we should contact Joachim so that he can try with that patch.

**Comment by sloriot** [comment]
I rebased the branch onto the 4.7 branch. See [here](https://github.com/sloriot/cgal/tree/4.7/Installation-split_config-pmoeller).

I tested:
* building a demo in branch build (`-DWITH_demos=ON` and alone)
* building a demo from a release created from this branch (`-DWITH_demos=ON`, alone and in-source)

When the merge in master will occur, `CMAKE_ALLOW_LOOSE_LOOP_CONSTRUCTS` should be removed from the newly created `CMakeLists.txt`.

**Comment by lrineau** [comment]
@bo0ts I have pushed the rebased version of Sébastien (see 
https://github.com/CGAL/cgal/pull/788#issuecomment-190152121), in the PR. If you want to work again on this branch, you will have to `reset --hard` or `rebase` your local branch.

**Comment by sloriot** [comment]
@joachim-reichel, could you check if this solves #471 ?

**Comment by joachim-reichel** [comment]
CGAL-4.8-Ic-150 does not work for me, same error message as in the original bug report (demo/Surface_modeling seems to be gone, but the examples have the same problem). I also do not see the mentioned files CGAL${LIBRARY_NAME}LibConfig.cmake.

Attached is a repro script that leaves out all the Debian packaging stuff. Start from the root directory of an unpacked tarball.
[repro.sh.txt](https://github.com/CGAL/cgal/files/157693/repro.sh.txt)



Repro case:

**Comment by sloriot** [comment]
@joachim-reichel, It was merged in CGAL-4.8-Ic-139

**Comment by sloriot** [comment]
@joachim-reichel I gave it a try with 4.8-Ic-139 with a modified version of your [script](https://gist.github.com/sloriot/663c40d41d539f0d3cdd) and it worked for me.
However, that script also worked on 4.8-Ic-140 which does not contain the fix. I'm probably missing something here.

**Comment by joachim-reichel** [comment]
Ok, I assumed that when it was merged in -139 all later version would contain that fix as well.  But even with -139 it does not work for me. The output ends with

-- Detecting CXX compiler ABI info - done
CMake Error at /tmp/cgal/install/lib/x86_64-linux-gnu/cmake/CGAL/CGAL_Qt5Exports.cmake:65 (message):
  The imported target "CGAL::CGAL_Qt5" references the file
     "/tmp/cgal/install/lib/x86_64-linux-gnu/libCGAL_Qt5.a"
  but this file does not exist.  Possible reasons include:
[...]
Call Stack (most recent call first):
  /tmp/cgal/install/lib/x86_64-linux-gnu/cmake/CGAL/CGALConfig.cmake:17 (include)
  CMakeLists.txt:10 (find_package)
-- Configuring incomplete, errors occurred!
See also "/tmp/cgal/test/examples/Surface_mesh/CMakeFiles/CMakeOutput.log".

**Comment by joachim-reichel** [comment]
Maybe libCGAL_Qt5.a was never configured/build in your test?

**Comment by sloriot** [comment]
Ah maybe I didn't get what should be tested. Let me give another try.

**Comment by lrineau** [comment]
@bo0ts I have pushed back the old value of your branch. The PR will eventually target 4.8. Too risky.

**Comment by sloriot** [comment]
Actually the previous code was correct, the extra patch simply do not include config and export when not needed. I merged that branch in tonight testsuite (4.8-Ic-151)

**Comment by joachim-reichel** [comment]
4.8-Ic-151 works fine, with the script and with actual Debian packages.

File: Installation/cmake/modules/CGALConfig_install.cmake.in [hunk 1]
```diff
@@ -38,52 +37,6 @@ set(CGAL_INCLUDE_DIRS  "@CMAKE_INSTALL_PREFIX@/@CGAL_INSTALL_INC_DIR@" )
...
-set(CGAL_Core_3RD_PARTY_DEFINITIONS    "@CGAL_Core_3RD_PARTY_DEFINITIONS@" )
-set(CGAL_Core_3RD_PARTY_LIBRARIES_DIRS "@CGAL_Core_3RD_PARTY_LIBRARIES_DIRS@" )
-set(CGAL_Core_3RD_PARTY_LIBRARIES      "@CGAL_Core_3RD_PARTY_LIBRARIES@" )
-
-set(CGAL_ImageIO_3RD_PARTY_INCLUDE_DIRS   "@CGAL_ImageIO_3RD_PARTY_INCLUDE_DIRS@" )
-set(CGAL_ImageIO_3RD_PARTY_DEFINITIONS    "@CGAL_ImageIO_3RD_PARTY_DEFINITIONS@" )
-set(CGAL_ImageIO_3RD_PARTY_LIBRARIES_DIRS "@CGAL_ImageIO_3RD_PARTY_LIBRARIES_DIRS@" )
-set(CGAL_ImageIO_3RD_PARTY_LIBRARIES      "@CGAL_ImageIO_3RD_PARTY_LIBRARIES@" )
-set(CGAL_ImageIO_USE_ZLIB                 "@CGAL_ImageIO_USE_ZLIB@" )
-
```
Comment (lrineau) [inline Installation/cmake/modules/CGALConfig_install.cmake.in]: This line `CGAL_ImageIO_USE_ZLIB` has no equivalent in the new split files!

Comment (lrineau) [inline Installation/cmake/modules/CGALConfig_install.cmake.in]: Fixed by PR #898.

**Pull request merged**